### ISSUE TITLE
intellisense for declarations from an `include()`

### ISSUE THREAD

**Issue by david-fong** [josetr/cmake-language-support#20]
I'm not sure to what degree declarations from `include()`s are already added to intellisense, but I've noticed some things not being added. For example,

- `FetchContent_Declare` after `include(FetchContent)`
- `generate_export_header` after `include(GenerateExportHeader)`
- `${CMAKE_INSTALL_INCLUDEDIR}` after `include(GNUInstallDirs)`

Are there things that make something like this difficult to support?

**Comment by SpaceIm** [comment]
Even with `cmake.languageSupport.enableFileAPI=True`,  CMake Tools extension installed, and after a successful CMake configuration, functions, macros & variables coming from an `include()` don't have any syntax highlighting nor autocompletion.

At least in `include()` command, there is an autocompletion & syntax highlighting of CMake bundled module files (but not for custom module files of a project, I would have expected something like traversing `CMAKE_MODULE_PATH` and look at additional CMake module files in these directories).

**Comment by alexchandel** [comment]
Please add this.  This would be very helpful.  [Per documentation](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_declare), `FetchContent_Declare()` takes "any of the download, update or patch options that the [ExternalProject_Add](https://cmake.org/cmake/help/latest/module/ExternalProject.html#command:externalproject_add)".  That should be very easy to add.

**Comment by MathiasMagnus** [comment]
Indeed. `cmake_dependent_option()` similarly from `include(CMakeDependentOption)`. However it would be awesome if documentation of custom packages with built CMake documentation could be displayed similarly, thinking of project-specific modules that have CMake-like documentation built, such as [rocm-cmake](https://github.com/RadeonOpenCompute/rocm-cmake) used throughout the ROCm ecosystem.

**Comment by billgan1024** [comment]
Any updates on this? I tried using `  "cmake.languageSupport.enableFileAPI": true` but functions obtained from `include()` are not highlighted until the include file is selected.
